lib/vsc/utils/zabbix.py

```diff
@@ -0,0 +1,64 @@
+
+# -*- encoding: utf-8 -*-
+#
+# Copyright 2012-2022 Ghent University
+#
+# This file is part of vsc-utils,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/hpcugent/vsc-utils
+#
+# vsc-utils is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Library General Public License as
+# published by the Free Software Foundation, either version 2 of
+# the License, or (at your option) any later version.
+#
+# vsc-utils is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU Library General Public License
+# along with vsc-utils. If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+This module adapts the nagios module so its output can be interpreted by Zabbix.
+
+@author: Samuel Moors (Vrije Universiteit Brussel)
+"""
+from __future__ import print_function
+
+import json
+import sys
+
+from vsc.utils.nagios import SimpleNagios, NagiosReporter
+from vsc.utils.script_tools import ExtendedSimpleOption
+
+
+class ZabbixReporter(NagiosReporter):
+    """Reporting class for Zabbix reports"""
+
+    def print_report_and_exit(self, timestamp, nagios_exit_code, nagios_exit_string, nagios_message):
+        """Print the zabbix report and exit"""
+        print('[%f, "%s", %s]' % (timestamp, nagios_exit_string, nagios_message))
+        self.log.info("Zabbix check cache file %s contents delivered: %s", self.filename, nagios_message)
+        sys.exit(nagios_exit_code)
```

Comment:
for zabbix, we want a normal exit I think? Will the agent not fail if it's non-zero?

Reply:
from Ward:

> Checking the exit code is removed from external checks, user parameters and system.run items. These checks will not to become unsupported if the exit code is different than zero. Exit code checks are now only performed in custom alert scripts, remote commands and user scripts executed on Zabbix server and proxy.

in de release notes van 3.4.3 

dus this ok